### PR TITLE
Event digester

### LINKED ISSUE #63: Aggregate digest tool for hourly data
Part of #24 involves simply counting events for each hour, which is a pretty decent-sized task, so it should be its own issue. It would be pretty crazy to just use aggregates over the entire collection.

Blocks #24

### PR DESCRIPTION
**Pull request by hut8**
Resolves #63

### CODE REVIEW

File: conf/build-ghc-app.sh [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #!/bin/bash
 set -e
 export GOPATH="${HOME}/go"
-export GHC_APP_DIR="/github-contributions/ghc-app"
```
Comment (joshjordan) [inline conf/build-ghc-app.sh]: Why did this go away?

File: conf/build-ghc-app.sh [hunk 2]
```diff
@@ -11,3 +10,8 @@ go get -d ./...
 export GOBIN='/srv/bin'
 go install ghc-app
 popd
+
+export EVENT_DIGEST_PATH="${GOPATH}/src/event-digest"
+pushd "${EVENT_DIGEST_PATH}" > /dev/null
```
Comment (joshjordan) [inline conf/build-ghc-app.sh]: Why to `/dev/null`?

File: conf/build-ghc-app.sh (hunk 1 above)
Comment (hut8) [inline conf/build-ghc-app.sh]: where is it referenced? I think I was planning on adding a symlink from there to somewhere else, but then that ended up in the provision.yml

File: conf/build-ghc-app.sh (hunk 2 above)
Comment (hut8) [inline conf/build-ghc-app.sh]: pushd and popd print useless messages; try it on your shell.

File: event-digest/digest_slice.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
+// Generated by: gen
+// TypeWriter: slice
+// Directive: +gen on *Digest
+
+package main
+
+// Sort implementation is a modification of http://golang.org/pkg/sort/#Sort
+// Copyright 2009 The Go Authors. All rights reserved.
```
Comment (joshjordan) [inline event-digest/digest_slice.go]: ?

**Comment by joshjordan** [comment]
How much of this did you write yourself? It might be useful to put a boundary (which might just be different files) that shows clearly what is vendor'd code and what is tenex code.

File: conf/build-ghc-app.sh (hunk 1 above)
Comment (joshjordan) [inline conf/build-ghc-app.sh]: So this change is unrelated to digests, yes?

File: conf/build-ghc-app.sh (hunk 2 above)
Comment (joshjordan) [inline conf/build-ghc-app.sh]: That is not true in my shell, but I do not have a reason to disbelieve you for another shell implementation.

```
/sites/fdn (jsj-private-firm-link) $ popd                                                                                   
/code/OpenEmu (master) $ pushd /sites/fdn                                                                                  
/sites/fdn (jsj-private-firm-link) $
```

**Comment by hut8** [comment]
I wrote most of it myself, except: clipperhouse.github.io/gen/

**Comment by joshjordan** [comment]
Good to `go`!

File: conf/build-ghc-app.sh (hunk 1 above)
Comment (hut8) [inline conf/build-ghc-app.sh]: yes, i was just editing this script and saw it was redundant
